monitor/src/main/java/edu/asu/diging/monitor/web/admin/ModifyRecipientController.java

```diff
@@ -0,0 +1,80 @@
+package edu.asu.diging.monitor.web.admin;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.monitor.core.exceptions.EmailAlreadyRegisteredException;
+import edu.asu.diging.monitor.core.exceptions.RecipientNotFoundException;
+import edu.asu.diging.monitor.core.model.INotificationRecipient;
+import edu.asu.diging.monitor.core.service.IAppManager;
+import edu.asu.diging.monitor.core.service.INotificationManager;
+import edu.asu.diging.monitor.web.admin.forms.AppForm;
+import edu.asu.diging.monitor.web.admin.forms.RecipientForm;
+
+@Controller
+public class ModifyRecipientController {
+
+    @Autowired
+    private INotificationManager manager;
+
+    @Autowired
+    private IAppManager appManager;
+
+    @RequestMapping(value = "/admin/recipients/modify", method = RequestMethod.POST)

```

Comment:
this request builds the page, right? That should not be a post request.

Reply:
I'm sending the email of the recipient (which is the id) as a parameter in the request body as sending it in the URL would not be correct ('cause we also have to send the csrf token in the request body). This is the reason I had to use POST here instead of GET. Could you let me know how I should go about this?